core/types/voters_test.go

```diff
@@ -0,0 +1,147 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/kowala-tech/kcoin/common"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"time"
+)
+
+func TestVoter_Properties(t *testing.T) {
+	address := common.Address{}
+	deposit := uint64(100)
+	weight := &big.Int{}
+	voter := NewVoter(address, deposit, weight)
+
+	assert.Equal(t, address, voter.Address())
+	assert.Equal(t, deposit, voter.Deposit())
+	assert.Equal(t, weight, voter.Weight())
+}
+
+func TestVoters_EmptyReturnsError(t *testing.T) {
+	voters, err := NewVoters(nil)
+
+	require.Error(t, err)
+	require.Nil(t, voters)
+}
+
+func TestVoters_GetAtNegativeIndexReturnsNil(t *testing.T) {
+	voter := makeVoter("0x5aaeb6053f3e94c9b9a09f33669435e7ef1beaed", 100, 100)
```

Comment:
you could create a testify suite and create a voter by default

Reply:
don't think here is the case, test suite is to for repetitive assertions? here this are normal objects not under test. 
Also the properties of Voter should be visible in the test scenario, like weight and deposit.

I did a global var with a list of voters, but don't think it reads better.

Your toughts?